modules/puppetca/puppetca_main.rb

```diff
@@ -47,11 +46,12 @@ def disable certname
     def autosign certname
       FileUtils.touch(autosign_file) unless File.exist?(autosign_file)
 
-      autosign = File.open(autosign_file, File::RDWR)
-      # Check that we don't have that host already
-      found = autosign.readlines.find { |line| line.chomp == certname }
-      autosign.puts certname unless found
-      autosign.close
+      open(autosign_file, File::RDWR) do |autosign|
+        # Check that we don't have that host already
+        found = false
+        autosign.each_line { |line| found = true if line.chomp == certname }
```

Comment:
I cannot remember why I chose that new approach. Maybe it was to assure myself that autosign pointer was at EOF. 

It works for me and has been in use since January.